test/widgets/recent_dm_conversations_test.dart

```diff
@@ -0,0 +1,329 @@
+      child: MaterialApp(
+        navigatorObservers: navigatorObserver != null ? [navigatorObserver] : [],
+        home: PerAccountStoreWidget(
+          accountId: eg.selfAccount.id,
+          child: const RecentDmConversationsPage()))));
+
+  // global store, per-account store, and page get loaded
+  await tester.pumpAndSettle();
+}
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  group('RecentDmConversationsPage', () {
+    Finder findConversationItem(Narrow narrow) {
+      return find.byWidgetPredicate(
+            (widget) =>
+        widget is RecentDmConversationsItem && widget.narrow == narrow,
+      );
+    }
+
+    testWidgets('page builds; conversations appear in order', (WidgetTester tester) async {
+      final user1 = eg.user(userId: 1);
+      final user2 = eg.user(userId: 2);
+
+      final message1 = eg.dmMessage(id: 1, from: eg.selfUser, to: [user1]); // 1:1
+      final message2 = eg.dmMessage(id: 2, from: eg.selfUser, to: []); // self-1:1
+      final message3 = eg.dmMessage(id: 3, from: eg.selfUser, to: [user1, user2]); // group
+
+      await setupPage(tester, users: [user1, user2],
+        dmMessages: [message1, message2, message3]);
+
+      final items = tester.widgetList<RecentDmConversationsItem>(find.byType(RecentDmConversationsItem)).toList();
+      check(items).length.equals(3);
+      check(items[0].narrow).equals(DmNarrow.ofMessage(message3, selfUserId: eg.selfUser.userId));
+      check(items[1].narrow).equals(DmNarrow.ofMessage(message2, selfUserId: eg.selfUser.userId));
+      check(items[2].narrow).equals(DmNarrow.ofMessage(message1, selfUserId: eg.selfUser.userId));
+    });
+
+    testWidgets('fling to scroll down', (WidgetTester tester) async {
+      final List<User> users = [];
+      final List<DmMessage> messages = [];
+      for (int i = 0; i < 30; i++) {
+        final user = eg.user(userId: i, fullName: 'User ${i.toString()}');
+        users.add(user);
+        messages.add(eg.dmMessage(id: i, from: eg.selfUser, to: [user]));
+      }
+
+      await setupPage(tester, users: users, dmMessages: messages);
+
+      final oldestConversationFinder = findConversationItem(
+        DmNarrow.ofMessage(messages.first, selfUserId: eg.selfUser.userId));
+
+      check(tester.any(oldestConversationFinder)).isFalse(); // not onscreen
+      await tester.fling(find.byType(RecentDmConversationsPage),
+        const Offset(0, -200), 4000);
+      await tester.pumpAndSettle();
+      check(tester.any(oldestConversationFinder)).isTrue(); // onscreen
+    });
+  });
+
+  group('RecentDmConversationsItem', () {
+    group('appearance', () {
+      void checkAvatar(WidgetTester tester, DmNarrow narrow) {
+        final shape = tester.widget<AvatarShape>(
+          find.descendant(
+            of: find.byType(RecentDmConversationsItem),
+            matching: find.byType(AvatarShape),
+          ));
+        check(shape)
+          ..size.equals(32)
+          ..borderRadius.equals(3);
```

Comment:
Makes sense. I'm not attached to it; I just saw it as an opportunity to be thorough. 🙂 I see what you mean about how this could get annoying if it weren't centralized in this helper.